lib/pavilion/plugins/results/filecheck.py

```diff
@@ -0,0 +1,56 @@
+from pathlib import Path
+from pavilion import config
+from pavilion import result_parsers
+from pavilion import utils
+import errno
+import glob
+import os
+import sys
+import yaml_config as yc
+
+
+class Filecheck(result_parsers.ResultParser):
+    """Checks the working directory for a given file.
+    The parser will tell the user if the filename exists or not. """
+
+    def __init__(self):
+        super().__init__(name='filecheck',
+                         description="Checks working directory"
+                         "for a given file")
+
+    def get_config_items(self):
+        # Result parser consists of 1 string elem: filename.
+        config_items = super().get_config_items()
+        config_items.extend([
+            yc.StrElem('filename', required=True,
+                       help_text="The filename result parser"
+                                 "takes in a string and checks"
+                                 "the working directory for said "
+                                 "file. It will return True with "
+                                 " the path-to-file or False.")
+        ])
+
+        return config_items
+
+    def _check_args(self, filename=None):
+        # Filename can being anything but none.
+        if not filename:
```

Comment:
you don't to check for  this if you indicated that this config element is required, which you did.

Source: [Config Element class in yaml_config](https://github.com/lanl-preteam/yaml_config/blob/92424cdf2d5f89f477d0fe74627ef9d4dcda6e30/lib/yaml_config/elements.py#L113)